crates/next-core/src/next_client_reference/ecmascript_client_reference/ecmascript_client_reference_module.rs

```diff
@@ -241,6 +241,11 @@ impl Module for EcmascriptClientReferenceModule {
 
         Ok(Vc::cell(references))
     }
+    #[turbo_tasks::function]
+    fn side_effects(self: Vc<Self>) -> Vc<ModuleSideEffects> {
+        // These just export some specially tagged functions
+        ModuleSideEffects::SideEffectFree.cell()
```

Comment:
This should be considered as side effect, otherwise

`import "client-component.js"` breaks when you assume it's eventually evaluated on client side. We had a bug report regarding that.